### PR TITLE
Add hooks nessie tests

### LINKED ISSUE #1522: Actions: nessie tests
- Validate output
- Test failure + output

### PR DESCRIPTION
**Pull request by itaiad200**
closes #1522

### CODE REVIEW

File: docs/assets/js/swagger.yml [hunk 1]
```diff
@@ -1909,6 +1909,40 @@ paths:
           schema:
             $ref: "#/definitions/error"
 
+  /repositories/{repository}/commits/{commitId}/runs:
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply docs/assets/js/swagger.yml:1912]: This is just a copy of the swagger file which is already merged.
Closing the comment and opening an issue for the discussion.

[the remark this answers is not in the document]
Comment (itaiad200) [reply docs/assets/js/swagger.yml:1912]: #1534

File: nessie/hooks_failure_test.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	docData := struct {
+		URL  string
+		Path string
+	}{
+		URL:  fmt.Sprintf("http://nessie:%d", server.port),
+		Path: path,
+	}
+
+	var doc bytes.Buffer
+	actionPreCommitTmpl := template.Must(template.New("action-pre-commit").Parse(actionPreCommitTemplate))
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply nessie/hooks_failure_test.go]: Done.

File: nessie/hooks_failure_test.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+	stats, err := client.Commits.Commit(
+		commits.NewCommitParamsWithContext(ctx).
+			WithRepository(repo).
+			WithBranch(branch).
+			WithCommit(&models.CommitCreation{Message: swag.String("Initial content")}),
+		nil)
+	require.Error(t, err, "commit should fail due to webhook")
+	require.Nil(t, stats)
+	require.Contains(t, err.Error(), "412")
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply nessie/hooks_failure_test.go]: Done.

File: nessie/webhook_server_test.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	return &webhookServer{
+		s:      s,
+		respCh: respCh,
+		port:   port,
+	}, nil
+}
+
+func timeoutHandlerFunc(chan hookResponse) func(http.ResponseWriter, *http.Request) {
+	return func(writer http.ResponseWriter, _ *http.Request) {
+		time.Sleep(2 * time.Minute)
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply nessie/webhook_server_test.go]: No need for fancy configuration, used the one we already have in-place instead.

**Review by nopcoder — APPROVED** [review]

**Pull request merged**